tests/test_bounce_integral.py

```diff
@@ -0,0 +1,703 @@
+    eq, result = eq.optimize(
+        objective=objective, constraints=constraints, optimizer=opt
+    )
+    print(result)
+
+    rho = np.array([0.5])
+    alpha = np.linspace(0, (2 - eq.sym) * np.pi, 10)
+    zeta = np.linspace(0, 6 * np.pi, 20)
+    bi, items = bounce_integral(
+        eq, rho=rho, alpha=alpha, zeta=zeta, return_items=True, check=True
+    )
+    B = items["data"]["B"]
+    pitch_res = 15
+    pitch = np.linspace(1 / B.max(), 1 / B.min(), pitch_res)
+    name = "g_zz"
+    f = eq.compute(name, grid=items["grid"], data=items["data"])[name]
+    result = bi(f, pitch)
+    assert np.isfinite(result).any(), "tanh_sinh quadrature failed."
+
+    # TODO now compare result to elliptic integral
+    bp1, bp2 = bounce_points(pitch, zeta, items["B.c"], items["B_z_ra.c"])
+
+
+@pytest.mark.unit
+def test_bounce_averaged_drifts():
+    """Test bounce-averaged drift with analytical expressions.
+
+    Calculate bounce-averaged drifts using the bounce-average routine and
+    compare it with the analytical expression
+    # Note 1: This test can be merged with the elliptic integral test as
+    we do calculate elliptic integrals here
+    # Note 2: Remove tests/test_equilibrium :: test_shifted_circle_geometry
+    # once all the epsilons and Gammas have been implemented and tested
+    """
+    eq = Equilibrium.load(".//tests//inputs//low-beta-shifted-circle.h5")
+
+    psi = 0.25  # rho^2 (or normalized psi)
+    alpha = 0
+
+    eq_keys = ["iota", "iota_r", "a", "rho", "psi"]
+
+    data_eq = eq.compute(eq_keys)
+
+    iotas = np.interp(np.sqrt(psi), data_eq["rho"], data_eq["iota"])
+    shears = np.interp(np.sqrt(psi), data_eq["rho"], data_eq["iota_r"])
+
+    N = int((2 * eq.M_grid) * 4 + 1)
+
+    zeta = np.linspace(-1.0 * np.pi / iotas, 1.0 * np.pi / iotas, N)
+    theta_PEST = alpha * np.ones(N, dtype=int) + iotas * zeta
+
+    coords1 = np.zeros((N, 3))
+    coords1[:, 0] = np.sqrt(psi) * np.ones(N, dtype=int)
+    coords1[:, 1] = theta_PEST
+    coords1[:, 2] = zeta
+
+    # Creating a grid along a field line
+    c1 = eq.compute_theta_coords(coords1)
+    grid = Grid(c1, sort=False)
+
+    # TODO: Request: The bounce integral operator should be able to take a grid.
+    #       Response: Currently the API is such that the method does all the
+    #                 above preprocessing for you. Let's test it for correctness
+    #                 first then do this later.
+    bi, items = bounce_integral(
+        eq,
+        rho=np.unique(coords1[:, 0]),
+        alpha=alpha,
+        zeta=zeta,
+        return_items=True,
+        check=True,
+    )
+    grid = items["grid"]
+    grid._unique_zeta_idx = np.unique(grid.nodes[:, 2], return_index=True)[1]
+
+    data_keys = [
+        "|grad(psi)|^2",
+        "grad(psi)",
+        "B",
+        "iota",
+        "|B|",
+        "B^zeta",
+        "cvdrift0",
+        "cvdrift",
+        "gbdrift",
+    ]
+
+    data = eq.compute(data_keys, grid=grid, data=items["data"])
```

Comment:
maybe it would be safer temporarily to not pass in the seed `data=items["data"]` and instead assign all the unique idx to the grid prior to calling eq.compute, as done in line 554.

Reply:
Can you give an example? I don't understand what you mean.

Reply:
It could have been, but I don't think it is anymore. I've pushed a commit to rule this out. Basically, under the assumption that eq.compute() always uses the correct grids to compute things as long as the grid has the necessary attributes it might be better to just tell it to start fresh and recompute everything (this is accomplished by not passing in the keyword argument `data=items["data"]`). Had lots of bugs in the past that came down using wrong grids etc. so I wanted to try this. Didn't seem to change anything.

Reply:
I get a discontinuous gbdrift even after removing data=items["data"] so there is definitely a discontinuity. 
I am doint eq.compute(data_keys, grid=grid)

Let me know if there's anything I can do.

Reply:
So we just need override_grid=False and this conversation can be resolved.